src/core/Object3D.js

```diff
@@ -538,44 +560,95 @@ THREE.Object3D.prototype = {
 
 	},
 
-	updateMatrix: function () {
-
-		this.matrix.compose( this.position, this.quaternion, this.scale );
-
-		this.matrixWorldNeedsUpdate = true;
-
-	},
-
-	updateMatrixWorld: function ( force ) {
-
-		if ( this.matrixAutoUpdate === true ) this.updateMatrix();
-
-		if ( this.matrixWorldNeedsUpdate === true || force === true ) {
-
-			if ( this.parent === undefined ) {
+	updateMatrix: ( function () {
+		
+		function updateMatrixWorldNeedsUpdate( o ) {
+			
+			o._matrixWorldNeedsUpdate = true;
+			
+		};
+		
+		return function() {
+
+			var position = this.position;
+			var quaternion = this.quaternion;
+			var scale = this.scale;
+			var _position = this._position;
+			var _quaternion = this._quaternion;
+			var _scale = this._scale;
+			
+			if ( ! _position.equals( position ) || ! _quaternion.equals( quaternion ) || ! _scale.equals( scale )  ) {
+
+				this._matrix.compose( position, quaternion, scale );
+				this.traverse( updateMatrixWorldNeedsUpdate );
+				_position.copy( position );
+				_quaternion.copy( quaternion );
+				_scale.copy( scale );
+			}
+			return this._matrix;
 
-				this.matrixWorld.copy( this.matrix );
+		};
 
-			} else {
+	} )(),
 
-				this.matrixWorld.multiplyMatrices( this.parent.matrixWorld, this.matrix );
+	updateMatrixWorld: function ( updateChildren, updateParents ) {
 
+		var parent = this.parent; 
+		
+		//check own matrix
+		if ( this.matrixAutoUpdate === true ) {
+			
+			this.updateMatrix();
+			
+		} else {
+			
+			this._matrixWorldNeedsUpdate = true;
+			
+		} 
+		
+		if ( parent === undefined ) {
+
+			// update own matrixWOrld
+			if ( this._matrixWorldNeedsUpdate === true ) {
+				
+				this._matrixWorld.copy( this._matrix );
+				
 			}
+			
+		} else {
+		
+			// first update parents before check matrixWorldNeedsUpdate
+			if ( updateParents !== false ) {
```

Comment:
Could we make this check `updateParents === true`? This will make `updateMatrixWorld()` a bit more intuitive. (I'm aware that most of the calls will become `.updateMatrixWorld( true, true )` and `.updateMatrixWorld( false, true )`.